lightning_client_test.go

```diff
@@ -0,0 +1,166 @@
+package lndclient
+
+import (
+	"context"
+	"errors"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/lightningnetwork/lnd/lnrpc"
+	"github.com/lightningnetwork/lnd/lnrpc/invoicesrpc"
+	"github.com/lightningnetwork/lnd/lntypes"
+	"github.com/lightningnetwork/lnd/lnwire"
+	"google.golang.org/grpc"
+)
+
+// addInvoiceArg records the args used in a call to mockRPCClient.AddInvoice.
+type addInvoiceArg struct {
+	in   *lnrpc.Invoice
+	opts []grpc.CallOption
+}
+
+// mockRPCClient implements lnrpc.LightningClient with dynamic method implementations and call spying.
+type mockRPCClient struct {
+	lnrpc.LightningClient
+
+	addInvoice     func(in *lnrpc.Invoice, opts ...grpc.CallOption) (*lnrpc.AddInvoiceResponse, error)
+	addInvoiceArgs []addInvoiceArg
+}
+
+func (m *mockRPCClient) AddInvoice(ctx context.Context, in *lnrpc.Invoice, opts ...grpc.CallOption) (*lnrpc.AddInvoiceResponse, error) {
+	m.addInvoiceArgs = append(m.addInvoiceArgs, addInvoiceArg{
+		in:   in,
+		opts: opts,
+	})
+
+	return m.addInvoice(in, opts...)
+}
+
+func Test_lightningClientAddInvoice(t *testing.T) {
```

Comment:
I've updated the function name to `TestLightningClientAddInvoice`, and have added a godoc comment for the test.

I believe the punctuation in this commit ("`// Set up the test structure.`") has been addressed since [f63e1a1](https://github.com/lightninglabs/lndclient/pull/107/commits/f63e1a18bc056524ad758d10f0e926b4b3084e7c), but feel free to reopen if that's not the case.